x/profile/internal/keeper/handler.go

```diff
@@ -0,0 +1,168 @@
+	if address := keeper.GetMonikerRelatedAddress(ctx, msg.Moniker); address.Equals(msg.Creator) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			fmt.Sprintf("An account with %s moniker already exists", msg.Moniker))
+	}
+
+	account := types.NewProfile(msg.Moniker, msg.Creator).
+		WithName(msg.Name).
+		WithSurname(msg.Surname).
+		WithBio(msg.Bio).
+		WithPictures(msg.Pictures)
+
+	// Before saving this method checks if an account with the same moniker exist
+	err := keeper.SaveProfile(ctx, account)
+	if err != nil {
+		return nil, err
+	}
+
+	createEvent := sdk.NewEvent(
+		types.EventTypeProfileCreated,
+		sdk.NewAttribute(types.AttributeProfileMoniker, account.Moniker),
+		sdk.NewAttribute(types.AttributeProfileCreator, account.Creator.String()),
+	)
+
+	ctx.EventManager().EmitEvent(createEvent)
+
+	result := sdk.Result{
+		Data:   keeper.Cdc.MustMarshalBinaryLengthPrefixed(account.Moniker),
+		Events: sdk.Events{createEvent},
+	}
+
+	return &result, nil
+}
+
+const (
+	defaultValue = "default"
+)
+
+// returns the profile with the proper edited fields
+// default string is used to let user replace previous inserted values with blank or empty ones
+func GetEditedProfile(account types.Profile, msg types.MsgEditProfile) types.Profile {
+	if msg.NewMoniker != defaultValue {
+		account.Moniker = msg.NewMoniker
+	}
+
+	if msg.Name != defaultValue {
+		account = account.WithName(msg.Name)
+	}
+
+	if msg.Surname != defaultValue {
+		account = account.WithSurname(msg.Surname)
+	}
+
+	if msg.Bio != defaultValue {
+		account = account.WithBio(msg.Bio)
+	}
+
+	if msg.ProfilePic != defaultValue && msg.ProfileCov != defaultValue {
+		pictures := types.NewPictures(msg.ProfilePic, msg.ProfileCov)
+		account = account.WithPictures(pictures)
+	} else {
+		if msg.ProfilePic != defaultValue && msg.ProfileCov == defaultValue {
+			account.Pictures.Profile = msg.ProfilePic
+		}
+		if msg.ProfileCov != defaultValue && msg.ProfilePic == defaultValue {
+			account.Pictures.Cover = msg.ProfileCov
+		}
+		account = account.WithPictures(account.Pictures)
+	}
+
+	return account
+}
+
+// handleMsgEditProfile handles the edit of a profile
+func handleMsgEditProfile(ctx sdk.Context, keeper Keeper, msg types.MsgEditProfile) (*sdk.Result, error) {
+	account, found := keeper.GetProfile(ctx, msg.Creator)
+	if !found {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			fmt.Sprintf("No existent profile to edit for address: %s", msg.Creator))
+	}
+	previousMoniker := account.Moniker
+
+	account = GetEditedProfile(account, msg)
+	err := account.Validate()
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	err = keeper.SaveProfile(ctx, account)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("An account with moniker: %s has already been created", msg.NewMoniker))
+	}
+
+	keeper.DeleteMonikerAddressAssociation(ctx, previousMoniker)
```

Comment:
I think this call should probably be done inside the `keeper.SaveProfile` method so that when that method is called, it handles all the cases properly.